src/Nix/Value.hs

```diff
@@ -43,6 +44,38 @@ import           Nix.Frames
 import           Nix.Scope
 import           Nix.Thunk
 import           Nix.Utils
+import           Data.Semigroup
+
+-- | A 'ContextFlavor' describes the sum of possible derivations for string contexts
+data ContextFlavor = 
+    DirectPath
+  | DerivationOutput
```

Comment:
`DerivationOutput` should take a `Text` (referring to the specific output requested, e.g. `"out"`)